### PR TITLE
Rework Using Native API Calls due to Library not being complete.

### PR DESCRIPTION
**Pull request by kvedder-amplex**

### CODE REVIEW

File: actions/lib/base.py [hunk 1]
```diff
@@ -1,38 +1,109 @@
...
+    "PowerDNSClient"
+
+]
+
+LOG = logging.getLogger(__name__)
+
+
+class PowerDNSClient(Action):
+
+    def __init(self, config):
```
Comment (blag) [inline actions/lib/base.py]:
```suggestion
    def __init__(self, config):
```

File: actions/lib/base.py [hunk 2]
```diff
@@ -1,38 +1,110 @@
...
+        # primary function used to talk to powerdns library
+
+        #setup api client
+        api_client = powerdns.PDNSApiClient(api_endpoint=self.config['api_url'], api_key=self.config['api_key'])
+        self.api = powerdns.PDNSEndpoint(api_client)
+        return self.api
+
+
+
+    # def __init__(self, config, timeout=5):
```
Comment (nzlosh) [inline actions/lib/base.py]: What's the reason for restructuring the entire base class and removing the base methods?

**Comment by kvedder-amplex** [comment]
Hello,
Made some substantial changes. I found the powerdns library to be incomplete.  I need this pack to allow to calls to multiple different servers so I added fields for this to the pack on each action. I know this is a huge deviation from previous designs. It's what I need at my workplace currently, but I figured I would submit it here also.

**Comment by blag** [comment]
Hey @kvedder-amplex, thanks for updating this pack, I highly appreciate the effort you've put into it so far.

Have you given any thoughts to keeping backwards compatibility? I'm sure there are existing users of this pack, and it may be a bit upsetting for them to have to change all of their workflows to match the new actions (once this is merged). Although, to that point, you will have a difficult time convincing us to merge this in if you do not attempt to maintain some amount of reverse compatibility, or if you do not attempt to justify why you cannot or should not have to maintain reverse compatibility.

Also, I'm not 100% sure that the `README.md` is successfully being rendered from `README.jinja`.

Finally, reviewing and merging large rewrites like this PR is normally a bit of a struggle. And the StackStorm team is highly resource constrained at the moment. So once you have a good idea of the scope of your rewrite, it might be easier to identify smaller chunks of it and submit them as separate PRs.

**Comment by kvedder-amplex** [comment]
I'll pass on this PR and use the pack internally. Hopefully the library gets published into PyPi someday and becomes production ready, and that the rest of the actions get built. Thanks for the good feedback, I'll keep it in mind going forward :)

**Pull request closed without merging**